GX-TimeFrequency/src/minimax_tau.F90

```diff
@@ -2907,50 +2906,36 @@ subroutine get_points_weights_tau(grid_size, e_range, ac_we, ierr)
     integer, intent(out)                                   :: ierr
 
     !> Internal variables
-    integer                                                :: ien, kloc, bup
+    integer                                                :: kloc, bup
     type(er_aw_aux)                                        :: aw
     real(kind=dp)                                          :: e_ratio
 
-
     !> Begin work
     ierr = 0
+    if (.not. any(tau_npoints_supported == grid_size)) then
+       ierr = 1
+       _REGISTER_EXC("The grid size you chose is not available.")
+       return
+    end if
 
-    if (any(tau_npoints_supported == grid_size)) then
-       ! Find location of grid size
-       kloc = findloc(tau_npoints_supported, grid_size, 1)
-       bup = energy_ranges_grids(kloc)
-
-       ! Allocate and set type elements
-       allocate(aw%energy_range(bup))
-       allocate(aw%aw_erange_matrix(2*grid_size, bup+1))
-       call set_aw_array(grid_size, aw)
-
-
-       ! Select energy region with binary search
-       ien = find_erange(bup, aw%energy_range, e_range)
-
-       ! Scale grids for large sizes when erange falls in the
-       ! first energy range
-       e_ratio = 1.0_dp
-       if (ien == 1 .and. grid_size > 20) then
-          e_ratio = aw%energy_range(1)/e_range
-          if (e_ratio > 1.5_dp) then
-             e_ratio = e_ratio/1.5_dp
-          endif
-       end if
+    ! Find location of grid size
+    kloc = findloc(tau_npoints_supported, grid_size, 1)
+    bup = energy_ranges_grids(kloc)
 
-       ac_we(:) = aw%aw_erange_matrix(:, ien)
+    ! Allocate and set type elements
+    allocate(aw%energy_range(bup))
+    allocate(aw%aw_erange_matrix(2*grid_size, bup+1))
+    call set_aw_array_tau(grid_size, aw)
 
-       ac_we(:) = ac_we(:) * e_ratio
+    ! Get coefficients and weights
+    e_ratio = 1.0_dp
+    call aw%get_coeff_weight(grid_size, bup, e_range, ac_we, e_ratio)
+    ac_we(:) = ac_we(:) * e_ratio
 
-       ! Deallocate
-       deallocate(aw%energy_range)
-       deallocate(aw%aw_erange_matrix)
-    else
-       ierr = 1
-       _REGISTER_EXC("The grid size you chose is not available.")
-    end if
+    ! Deallocate
+    deallocate(aw%energy_range)
+    deallocate(aw%aw_erange_matrix)
```

Comment:
These would get replaced with a single call to the destructor (see prior comments)

`call aw%finalze()`